nixos/modules/services/misc/ersatztv.nix

```diff
@@ -0,0 +1,114 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+
+let
+  inherit (lib)
+    mkIf
+    getExe
+    maintainers
+    mkEnableOption
+    mkOption
+    mkPackageOption
+    ;
+  inherit (lib.types) str path bool;
+  cfg = config.services.ersatztv;
+in
+{
+  options = {
+    services.ersatztv = {
+      enable = mkEnableOption "ErsatzTV";
+
+      package = mkPackageOption pkgs "ersatztv" { };
+
+      user = mkOption {
+        type = str;
+        default = "ersatztv";
+        description = "User account under which ErsatzTV runs.";
+      };
+
+      group = mkOption {
+        type = str;
+        default = "ersatztv";
+        description = "Group under which ErsatzTV runs.";
+      };
+
+      stateDir = mkOption {
+        type = path;
+        default = "/var/lib/ersatztv";
+        description = ''
+          State directory for ErsatzTV
+        '';
+      };
+
+      baseUrl = mkOption {
```

Comment:
Rather than a `baseUrl` option, I would add an `environment`/`settings` option, with default values for the following:

* [`ETV_UI_PORT`](https://ersatztv.org/docs/networking/#port-numbers): 8409
* `ETV_BASE_URL`: `/`

Reply:
Unfortunately upstream does not document their environment variables very well. This should be brought up as an issue with them and not hold up merging this module IMO.